### PR TITLE
[Wallet] Prefer coins that have fewer ancestors, sanity check txn before ATMP and rebroadcast functional test

### PR DESCRIPTION
**Pull request by furszy**
Work originated from a third party service hitting the mempool ancestors count limit.

Focused on two main topics (plus some extra cleanup):

1) The wallet's utxo selection process will now predilect coins that have fewer ancestors.
2) Add functional test for the wallet transactions rebroadcast.

Based on the following backports:

* #6669 (only 6b0e622c).
* #9262.
* And ported the wallet transactions rebroadcast functional test, skipping and grouping few intermediate PRs up to faf6f156.

#### Important Note: 
This PR is not meant to be included in the shortly coming v5.3.2.1. Which is good enough as is, the `sendmany` min depth arg fix is already by-passing the issue automatically without human interaction (mempool utxo will not be selected by default).
This is a follow up work to fully prevent the issue from happening in other flows and not only in `sendmany`.
Merge target is v5.4.

### CODE REVIEW

**Comment by random-zebra** [comment]
I think that the introduction of  `-walletrejectlongchains` argument is redundant, due to our current behavior (which diverges from upstream's).
The wallet should *always* reject long chains, as any transaction that fails to commit to a node's mempool would be abandoned by the node's wallet (#1695).
Due to this, the last extra run of `SelectCoinsMinConf` is also redundant.

**Comment by furszy** [comment]
good catch zebra ☕, true that is redundant. Squashed changed in 503ce57

**Comment by furszy** [comment]
rebased on master.

**Review by Fuzzbawls — APPROVED** [review]
ACK 26dcd16c7b353b9921a427f3178392be30d1575c

File: test/functional/wallet_basic.py [hunk 1]
```diff
@@ -224,6 +225,26 @@ def run_test(self):
...
+        # Get all non-zero utxos together
+        chain_addrs = [self.nodes[0].getnewaddress(), self.nodes[0].getnewaddress()]
+        singletxid = self.nodes[0].sendtoaddress(chain_addrs[0], self.nodes[0].getbalance(), "", "", True)
+        self.nodes[0].generate(1)
+        node0_balance = self.nodes[0].getbalance()
+        # Split into two chains
+        rawtx = self.nodes[0].createrawtransaction([{"txid":singletxid, "vout":0}], {chain_addrs[0]:node0_balance/2-Decimal('0.01'), chain_addrs[1]:node0_balance/2-Decimal('0.01')})
+        signedtx = self.nodes[0].signrawtransaction(rawtx)
+        singletxid = self.nodes[0].sendrawtransaction(signedtx["hex"])
+        self.nodes[0].generate(1)
```
Comment (random-zebra) [inline test/functional/wallet_basic.py:239]: would be good to add an `assert_equal(len(self.nodes[0].listunspent()), 2)` here, just to be triple sure.

**Review by random-zebra — COMMENTED** [review]
Need to add the new test `wallet_resendwallettransactions.py` to the test runner file (in the "less than 60s" list).

File: test/functional/wallet_basic.py (hunk 1 above)
Comment (furszy) [reply test/functional/wallet_basic.py:239]: squashed inside 266cc9e

**Comment by furszy** [comment]
yes good, done, new test added to the test runner. Squashed in 5cd487e

**Review by random-zebra — APPROVED** [review]
ACK 9deb391a9dc642d4a5bbfc495919a96a12ae8f51

**Pull request merged**